R/utils.R

```diff
@@ -259,84 +259,333 @@ rowcol_to_dim <- function(row, col) {
   # we will always return something like "A1"
   stringi::stri_join(min_col, min_row)
 }
-#' @rdname dims_helper
-#' @param ... construct dims arguments, from rows/cols vectors or objects that can be coerced to data frame
+#' Helper to specify the `dims` argument.
+#'
+#' @description
+#' `wb_dims()` can be used to help provide the `dims` argument, in the `wb_add_*` functions.
+#' It returns a Excel range (i.e. "A1:B1") or a start like "A2".
+#' It can be very useful as you can specify many parameters that interact together
+#' In general, you must provide named arguments. `wb_dims()` will only accept unnamed arguments
+#' if it is `rows`, `cols`, for example `wb_dims(1:4, 1:2)`, that will return "A1:B4".
+#'
+#' `wb_dims()` can also be used with an object (a `data.frame` or a `matrix` for example.)
+#' All parameters are numeric unless stated otherwise.
+#' # Using `wb_dims()` alone
+#'
+#' * `rows` / `cols` (if you want to specify a single one, use `start_row` / `start_col`)
+#' * `start_row`
+#' * `start_col`
+#'
+#' # Using `wb_dims()` with an object
+#'
+#' * `x` An object (typically a `matrix` or a `data.frame`)
+#' * `start_row` the starting row of `x` (The `dims` returned will be )
+#' * `start_col` the starting column: a single integer, or an Excel column identifier "A", "B" etc.
+#' * `rows` (Which rows? (not fully supported yet
+#' * `cols` a range of column, or one of the column names of `x` (length 1 only accepted)
+#' * `row_names` A logical, should include `row_names`
+#' * `col_names` A logical, defaults to `TRUE` is `x` has dimensions.
+#'          Using `FALSE` can be useful to apply a style or a formula to the content of `x`.
+#' @return A `dims` string
+#' @param ... construct dims arguments, from rows/cols vectors or objects that
+#'   can be coerced to data frame
 #' @examples
-#' # either vectors
+#' # Provide coordinates
+#' wb_dims()
+#' wb_dims(1, 4)
+#' wb_dims(rows = 1, cols = 4)
+#' wb_dims(start_row = 4)
+#' wb_dims(start_col = 2)
```

Comment:
Maybe `start_col`/`start_row` was a bad initial idea. I thought of it like `to_col`/`to_row` or `from_col`/`from_row`.

Reply:
If you replace everywhere start_row, by from_row, you see that it's exactly the mindset I had! Could rename? but that would mean a new param for user (that is not used in the rest of openxlsx2) 
this could be addressed in docs ?

Reply:
Guess it is preferable to have arguments that are self explanatory. So I'd go for it

Reply:
Ok, will do, should `start_row`, `start_col` error? or only throw a warning?

Reply:
Right now it's just in development, therefore we can simply replace. But error is fine too

Reply:
Renamed it `from_col` / `from_row` with a special error message if `start_row` / `start_col` are provided.